elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +267,382 @@ private Object fetchRelationship(Environment context) {
+                attributesOnly.put(key, entry.get(key));
+            }
+        }
+        return attributesOnly;
+    }
+
+    /**
+     * Creates a new "root-level" resource in the database
+     * @param input individual input data objects
+     * @param outputType graphql-java output type
+     * @param fieldName graphql-java field type
+     * @param requestScope request scope
+     * @param uuid uuid
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource createObject(Map<String, Object> input, String fieldName,
+                                RequestScope requestScope, Optional<String> uuid,
+                                GraphQLType outputType, PersistentResource parent) {
+        EntityDictionary dictionary = requestScope.getDictionary();
+
+        Class<?> entityClass;
+        if(parent == null) {
+            GraphQLObjectType objectType = (GraphQLObjectType) ((GraphQLList) outputType).getWrappedType();
+            entityClass = dictionary.getEntityClass(objectType.getName());
+        } else {
+            entityClass = dictionary.getParameterizedType(parent.getResourceClass(), fieldName);
+        }
+        PersistentResource toCreate = PersistentResource.createObject(parent, entityClass, requestScope, uuid);
+
+        //add new relation between parent and the updated object
+        if(parent != null) {
+            parent.addRelation(fieldName, toCreate);
+        }
+        return updateObject(new HashSet<>(Arrays.asList(toCreate)), input, fieldName, requestScope, parent);
+    }
+
+    /**
+     * Updates an object
+     * @param toUpdate entities to update
+     * @param input input data to update entities with
+     * @return updated object
+     */
+    private PersistentResource updateObject(Set<PersistentResource> toUpdate, Map<String, Object> input,
+                                String fieldName, RequestScope requestScope,
+                                            PersistentResource parent) {
+        EntityDictionary dictionary = requestScope.getDictionary();
+        Class<?> entityClass;
+        if(parent == null) {
+            entityClass = dictionary.getEntityClass(fieldName);
+        } else {
+            entityClass = dictionary.getParameterizedType(parent.getResourceClass(), fieldName);
+        }
+
+        String idFieldName = dictionary.getIdFieldName(entityClass);
+        PersistentResource obj = toUpdate.iterator().next();
+        for(Map.Entry<String, Object> row : input.entrySet()) {
+            if(dictionary.isAttribute(entityClass, row.getKey())) { /* iterate through each attribute provided */
+                obj.updateAttribute(row.getKey(), row.getValue());
+            } else if(Objects.equals(row.getKey(), idFieldName)) { /* update 'id' attribute */
+                obj.updateAttribute(row.getKey(), row.getValue());
+            } else {
+                throw new BadRequestException("Unrecognized attribute passed to 'data': " + row.getKey());
```

Comment:
I would throw an IllegalStateException.   This should never happen (and would require a bug in your code somewhere).  A BadRequestException is meant to be propagated back to the client as a 400 error (they did something wrong).  Instead, this should return a 500 error (we did something wrong).